packages/vm/src/runTx.ts

```diff
@@ -619,6 +671,20 @@ async function _runTx(this: VM, opts: RunTxOpts): Promise<RunTxResult> {
     }
   }
 
+  /**
+   * Cleanup code of accounts written to in a 7702 transaction
+   */
+
+  for (const str of writtenAddresses) {
+    const address = Address.fromString(str)
+    // Special case: in StateManager if we try to set code to the empty code, we ignore this
```

Comment:
Not directly part of this PR but why is this that we don't update the code hash in the state manager if it's `KECCAK_NULL`?

Reply:
I think the reasoning is that if we try to set code to `KECCAK_NULL` this is a no-op, since the code of an account is empty by default. Note that once code is set, the only way to (currently) remove it is to use SELFDESTRUCT which calls another method `deleteAccount`.

However, thus now this invariant is broken by this EIP, since we first set code, and then unset it, and this is not done by SELFDESTRUCT.